prog/vnet/nic_nexus.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::NicNexus < Prog::Base
+  semaphore :destroy, :refresh_mesh, :detach_vm
+
+  def self.assemble(private_subnet_id, name: nil, ipv6_addr: nil, ipv4_addr: nil)
+    unless (subnet = PrivateSubnet[private_subnet_id])
+      fail "Given subnet doesn't exist with the id #{private_subnet_id}"
+    end
+
+    ubid = UBID.generate(UBID::TYPE_NIC)
+    name ||= Nic.ubid_to_name(ubid)
+
+    Validation.validate_name(name)
```

Comment:
Interesting, should this be pushed down in a separate patch? are there situations we don't want a "valid" name?

Reply:
Yeah, characters in resource name may result in our paths in the console not usable. I can leave this to the UI PR though.